experimental/iterators/lib/Conversion/IteratorsToLLVM/IteratorAnalysis.cpp

```diff
@@ -99,6 +99,24 @@ LLVMStructType StateTypeComputer::operator()(
       op->getContext(), "iterators.reduce_state", {upstreamStateTypes[0]});
 }
 
+/// The state of TabularViewToStreamOp consists of a single number that
+/// corresponds to the index of the next struct returned by the iterator and the
+/// input tabular view. Pseudo-code:
+///
+/// template <typename TabularViewType>
+/// struct { int64_t currentIndex; TabularViewType view; }
+template <>
+LLVMStructType StateTypeComputer::operator()(
+    TabularViewToStreamOp op,
+    llvm::SmallVector<LLVM::LLVMStructType> /*upstreamStateTypes*/) {
+  MLIRContext *context = op->getContext();
+  Type i64 = IntegerType::get(context, /*width=*/64);
+  Type viewType = typeConverter.convertType(op.input().getType());
+  return LLVM::LLVMStructType::getNewIdentified(
```

Comment:
Same here re LLVM dependence.

What is tricky here is that you already seem to depend on the LLVM dialect for this.

We can keep this as a later TODO but please be sure to cleanup before embarking on more abstractions.